plugins/texteditor.koplugin/main.lua

```diff
@@ -0,0 +1,160 @@
+                        self.file_path = ""
+                        UIManager:close(self.input)
+                    end
+                })
+
+            end,
+        },
+        {
+            text = _("Pg_up"),
+            callback = function()
+                self.input._input_widget.text_widget:scrollUp()
+                UIManager:setDirty(self.input, function()
+                    return "ui", self.input.dimen
+                end)
+            end,
+        },
+        {
+            text = _("Pg_dwn"),
+            callback = function()
+                self.input._input_widget.text_widget:scrollDown()
+                UIManager:setDirty(self.input, function()
+                    return "ui", self.input.dimen
+                end)
+            end,
+        }
+
+    }},
+    }
+    UIManager:show(self.input)
+    self.input:onShowKeyboard()
+end
+
+
+function TextEditor:chooseFile()
+    self.input:onClose()
+    local path_chooser = PathChooser:new{
+        title = _("Choose file. Long press to confirm"),
+        height = Screen:getHeight(),
+        path = util.realpath(DataStorage:getDataDir()),
+        show_hidden = G_reader_settings:readSetting("show_hidden"),
+        file_filter = function() return true end,
+        onConfirm = function(file_path)
+            logger.dbg("TextEditor: selected file_file_path " .. file_path )
+            self.file_path = file_path
+            self:readFile(self.file_path)
+            self.input:onShowKeyboard()
+        end
+    }
+    UIManager:show(path_chooser)
+end
+
+function TextEditor:readFile(file_path)
+    logger.dbg("TextEditor: reading file: " .. file_path)
+    local file = assert(io.open(file_path, "rb"))
+    if file then
+        self.file_path = file_path
+        self.context = file:read("*all")
+        self.input:setInputText(self.context)
+        file:close()
+    else
+        UIManager:show(InfoMessage:new{
+            text = _("Failed to read file: \n" .. file_path)
+        })
+    end
+    UIManager:forceRePaint()
+end
+
+function TextEditor:saveFile(file_path)
+    logger.dbg("TextEditor: saving file: " .. file_path)
+    local file = io.open(file_path, "w")
+    if file then
+        UIManager:show(ConfirmBox:new{
+            text = _("Are you sure that you want to save changes to file: \n" .. file_path),
+            ok_text = _("Save"),
+            ok_callback = function()
+                file:write(self.input:getInputText())
+                file:close()
+                UIManager:show(InfoMessage:new{
+                    text = _("Saved file: " .. file_path)
+                })
+            end
+        })
+    else
+        UIManager:show(InfoMessage:new{
+            text = _("Failed to save file: \n" .. file_path)
+        })
+    end
+    UIManager:forceRePaint()
+end
+
+function TextEditor:addToMainMenu(menu_items)
+    menu_items.text_editor = {
+        text = _("Basic text editor"),
+        callback = function()
```

Comment:
It doesn't work if you use `callback = self.start,`?

Reply:
when I change it to ` callback = self.start,` then I've :
```
./luajit: plugins/texteditor.koplugin/main.lua:45: attempt to call method 'chooseFile' (a nil value)
stack traceback:
        plugins/texteditor.koplugin/main.lua:45: in function 'callback'
        frontend/ui/widget/button.lua:201: in function 'action'
        frontend/ui/uimanager.lua:597: in function '_checkTasks'
        frontend/ui/uimanager.lua:818: in function 'handleInput'
        frontend/ui/uimanager.lua:904: in function 'run'
        ./reader.lua:195: in main chunk
        [C]: at 0x00404960

```

Reply:
Ok